src/App.tsx

```diff
@@ -15,36 +16,113 @@
+
+  if (sortType) {
+    sortedData.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.ABC:
+          return good1.localeCompare(good2);
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+        case SortType.RESET:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    sortedData.reverse();
+  }
+
+  return sortedData;
+};
+
 export const App: React.FC = () => {
+  const [selectedSortType, setSelectedSortType] = useState<SortType | null>(
+    null,
+  );
+  const [isReversed, setReversed] = useState<boolean>(false);
+
+  const visibleData = handleSorting(
+    goodsFromServer,
+    selectedSortType,
+    isReversed,
+  );
+
+  function handleReset() {
+    setReversed(false);
+    setSelectedSortType(null);
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
-          Sort alphabetically
+        <button
+          type="button"
+          className={cn('button', {
+            'is-info': selectedSortType === SortType.ABC,
+            'is-light': selectedSortType !== SortType.ABC,
+          })}
+          onClick={() => setSelectedSortType(SortType.ABC)}
+        >
+          {SortType.ABC}
         </button>
 
-        <button type="button" className="button is-success is-light">
-          Sort by length
+        <button
+          type="button"
+          className={cn('button', {
+            'is-success': selectedSortType === SortType.LENGTH,
+            'is-light': selectedSortType !== SortType.LENGTH,
+          })}
+          onClick={() => setSelectedSortType(SortType.LENGTH)}
+        >
+          {SortType.LENGTH}
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button', {
+            'is-warning': isReversed,
+            'is-light': !isReversed,
+          })}
+          onClick={() => setReversed(prevState => !prevState)}
+        >
           Reverse
         </button>
-
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(selectedSortType || isReversed) && (
+          <button
+            type="button"
+            className={cn('button', {
+              'is-danger': selectedSortType === SortType.RESET,
+              'is-light': selectedSortType !== SortType.RESET,
+            })}
+            onClick={handleReset}
+          >
+            {SortType.RESET}
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {visibleData.map(good => (
+          <li data-cy="Good" key={good}>
```

Comment:
Using the `good` string as a key in a list is not reliable if the list could contain duplicate items or if the items can change. It's better to use a unique identifier for each item, such as an ID.